File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithInheritanceComponentsHelper.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.common.annotation.Root;

public interface AppWithInheritanceComponentsHelper {
    record Class1(Class2 class2) {}

    record Class2(Class3 class3) {}

    class Class3 {}

    interface AppWithInheritanceComponentsModule1 {
        @Root
        default Class1 class1(Class2 class2) {
            return new Class1(class2);
        }
    }

    interface AppWithInheritanceComponentsModule2 {
        default Class2 class2(Class3 class3) {
            return new Class2(class3);
        }
    }

    interface AppWithInheritanceComponentsModule3 {
        default Class3 class3() {
            return new Class3();
        }
    }
}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithInheritanceComponentsHelper.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.common.annotation.Root

interface AppWithInheritanceComponentsHelper {
    data class Class1(val class2: Class2)

    data class Class2(val class3: Class3)

    class Class3
    interface AppWithInheritanceComponentsModule1 {
        @Root
        fun class1(class2: Class2): Class1 {
            return Class1(class2)
        }
    }

    interface AppWithInheritanceComponentsModule2 {
        fun class2(class3: Class3): Class2 {
            return Class2(class3)
        }
    }

    interface AppWithInheritanceComponentsModule3 {
        fun class3(): Class3 {
            return Class3()
        }
    }
}
